### PR TITLE
test: enhance unit tests for keyUtils with comprehensive cases

### PR DESCRIPTION
**Pull request by abdulazeem-tk4vr**
### **User description**
This commit expands the test suite for the keyUtils utility functions, specifically focusing on the keyListAsLeveledKeys function. It includes tests for various scenarios such as transforming key lists, handling empty and null inputs, managing duplicate keys, and accommodating special characters and long key names. The enhancements ensure robust validation of key transformations and security level assignments, improving overall test coverage and reliability.

test : add more tests

___

### **PR Type**
Tests

___

### **Description**
- Enhance unit tests for `keyListAsLeveledKeys` function.

- Add tests for handling empty, null, and undefined inputs.

- Include tests for duplicate keys and special characters.

- Validate handling of various security levels and long key names.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>keyUtils.test.ts</strong><dd><code>Comprehensive test enhancements for `keyListAsLeveledKeys`</code></dd></summary>
<hr>

test/unit/src/utils/keyUtils.test.ts

<li>Added comprehensive test cases for <code>keyListAsLeveledKeys</code>.<br> <li> Included tests for empty, null, and undefined inputs.<br> <li> Tested handling of duplicate keys and special characters.<br> <li> Validated various security levels and long key names.

</details>

  </td>
  <td><a href="https://github.com/shardeum/shardeum/pull/402/files#diff-d19cd4f05a754a55e76431073b960a2db24d620de8d5232a83c976d9acb79d7e">+173/-22</a></td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

**Review by urnotsam — DISMISSED** [review]

File: test/unit/src/utils/keyUtils.test.ts [hunk 1]
```diff
@@ -1,39 +1,190 @@
...
+    /**
+     * Tests that different security levels can be assigned
+     */
+    it('should assign the specified security level to each key', () => {
+      // Arrange
+      const keyList = ['keyA', 'keyB']
+      const securityLevel = 5
+
+      // Act
+      const result = keyListAsLeveledKeys(keyList, securityLevel)
```
Comment (aniketdivekar) [inline test/unit/src/utils/keyUtils.test.ts]: This shouldn't be possible right? we have 0, 1, 2, 3 as values in `DevSecurityLevel` enum? 
This is debatable, but if we are not supporting a certain level in our enum, then we shouldn't be able to assign it.
It's possible that it's prevented somewhere early in the flow and this util never receives 5 as security level. 
My suggestion would be comment this out with a note that, this needs to be handled in main business logic if not done already. 
cc @BelfordZ @afostr

**Review by mhanson-github — APPROVED** [review]
